app/src/main/java/nu/yona/app/ui/dashboard/CustomPageAdapter.java

```diff
@@ -106,6 +121,25 @@ private ViewGroup initiateDayActivityReport(ViewGroup collection, int position)
         goalType = (YonaFontTextView) spreadView.findViewById(R.id.goalType);
         goalScore = (YonaFontTextView) spreadView.findViewById(R.id.goalScore);
         mSpreadGraph = (SpreadGraph) spreadView.findViewById(R.id.spreadGraph);
+
+        // This we have considered as an List of link will be display to user as per displaying in
```

Comment:
can't we declare all the List and layout manager previously and just notify adapter from here. 
as this logic will get called each time unnecessarily.

Reply:
This must be execute for each ViewPager page it load. Other thing is this is also the place where all view child get initialised and return as Parent view object.